scripts/copy.pl

```diff
@@ -107,31 +107,30 @@ sub cmd_num {
 		return();
 	}
 
-	$args = [split /[\s-]/, $args];
-	if(@$args > 2 || grep {/[^\d]+/} @$args) {
+	my @arg = [split /[\s-]/, $args];
+	if(@arg > 2 || grep {/[^\d]+/} @arg) {
 		Irssi::print('Illegal range!', MSGLEVEL_CLIENTCRAP);
 		return();
 	}
 
-	$args->[0] = 1 if ($args->[0]==0);
-	$args->[0] -= 1;
-	$args->[1] -= 1 if defined $args->[1];
+	$arg[0] = 1 if ($arg[0]==0);
+	$arg[0] -= 1;
+	$arg[1] -= 1 if defined $arg[1];
 
-	for(1..$args->[0]) {
+	for(1..$arg[0]) {
 		last unless $line->prev;
 		$line = $line->prev;
         }
 
         my $str;
-	for($args->[0]..($args->[1] // $args->[0])) {
-		$str = join "\n", $line->get_text(0), $str;
+	for($arg[0]..($arg[1] // $arg[0])) {
+		$str = join +(Irssi::settings_get_str($IRSSI{name}.'_file_eol') || "\n"), $line->get_text(0), $str;
```

Comment:
Irssi::settings_get_str($IRSSI{name}.'_file_eol') == '\n'

$copy_file_eol == "\n"

Reply:
I missed the global, fixed